src/Control/Carrier/FossaAPIClientIO.hs

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE GADTs #-}
+
+module Control.Carrier.FossaAPIClientIO (FossaAPIClientIOC, runFossaAPIClientIO) where
+
+import App.Fossa.FossaAPIV1 qualified as API
+import Control.Carrier.Simple
+import Control.Effect.Diagnostics (Diagnostics)
+import Control.Effect.FossaAPIClient
+import Control.Effect.Lift (Lift, sendIO)
+import Effect.Logger
+
+-- | A carrier to run Fossa API functions in the IO monad
+type FossaAPIClientIOC = SimpleC FossaAPIClientF
+
+-- | Runs FossaAPI effects as IO operations
+-- TODO: Put ApiOpts in a reader
+-- TODO: Exception handling
+runFossaAPIClientIO ::
+  ( Has (Lift IO) sig m
+  , Has Diagnostics sig m
+  , Has Logger sig m
+  ) =>
```

Comment:
Let's add the `ReaderC ApiOpts` here now, and see if we can improve the testing api from that.

Reply:
I think with some work, we can make `withMockApi` a lot easier to use, and more powerful.